src/test/java/org/apache/maven/shared/artifact/filter/OldPatternIncludesArtifactFilter.java

```diff
@@ -0,0 +1,428 @@
+        else if ( pattern.startsWith( "*" ) && pattern.endsWith( "*" ) )
+        {
+            final String contains = pattern.substring( 1, pattern.length() - 1 );
+
+            matches = token.contains( contains );
+        }
+        // support leading wildcard
+        else if ( pattern.startsWith( "*" ) )
+        {
+            final String suffix = pattern.substring( 1 );
+
+            matches = token.endsWith( suffix );
+        }
+        // support trailing wildcard
+        else if ( pattern.endsWith( "*" ) )
+        {
+            final String prefix = pattern.substring( 0, pattern.length() - 1 );
+
+            matches = token.startsWith( prefix );
+        }
+        // support wildcards in the middle of a pattern segment
+        else if ( pattern.indexOf( '*' ) > -1 )
+        {
+            String[] parts = pattern.split( "\\*" );
+            int lastPartEnd = -1;
+            boolean match = true;
+
+            for ( String part : parts )
+            {
+                int idx = token.indexOf( part );
+                if ( idx <= lastPartEnd )
+                {
+                    match = false;
+                    break;
+                }
+
+                lastPartEnd = idx + part.length();
+            }
+
+            matches = match;
+        }
+        // support versions range
+        else if ( pattern.startsWith( "[" ) || pattern.startsWith( "(" ) )
+        {
+            matches = isVersionIncludedInRange( token, pattern );
+        }
+        // support exact match
+        else
+        {
+            matches = token.equals( pattern );
+        }
+
+        return matches;
+    }
+
+    private boolean isVersionIncludedInRange( final String version, final String range )
+    {
+        try
+        {
+            return VersionRange.createFromVersionSpec( range ).containsVersion( new DefaultArtifactVersion( version ) );
+        }
+        catch ( final InvalidVersionSpecificationException e )
+        {
+            return false;
+        }
+    }
+
+    /** {@inheritDoc} */
+    public void reportMissedCriteria( final Logger logger )
+    {
+        // if there are no patterns, there is nothing to report.
+        if ( !positivePatterns.isEmpty() || !negativePatterns.isEmpty() )
+        {
+            final List<String> missed = new ArrayList<>();
+            missed.addAll( positivePatterns );
+            missed.addAll( negativePatterns );
+
+            missed.removeAll( patternsTriggered );
+
+            if ( !missed.isEmpty() && logger.isWarnEnabled() )
+            {
+                final StringBuilder buffer = new StringBuilder();
+
+                buffer.append( "The following patterns were never triggered in this " );
+                buffer.append( getFilterDescription() );
+                buffer.append( ':' );
+
+                for ( String pattern : missed )
+                {
+                    buffer.append( "\no  '" ).append( pattern ).append( "'" );
```

Comment:
Same here

Reply:
I prefer a platform independent output. This isn't just for stdout.

Now that I look at this line, I am confused what that "o" after \n is for.

Reply:
Even if it is for logging, your will always have mixed endings in your ouput. this is ugly.

Reply:
See https://github.com/apache/maven-common-artifact-filters/pull/15#issuecomment-766796366
I'm trying to minimise the refactoring in my commits, so don't blame me because the current code is not up to your standards :-)

Reply:
@elharo ... and the answer to your question is for presenting as a bullet point list:
```
o   first item
o   second item
...
```

Reply:
If this is in the current code, you don't have to fix it in this PR. In general, bullet points should be \u8226 or an asterisk, not the letter o. Hardly the most egregious problem in our code base of course.

Reply:
asterisks are good